src/polymer/polymer-element.ts

```diff
@@ -239,53 +239,152 @@ function propertyToAttributeName(propertyName: string): string|null {
+    } else {
+      if (superElements.size > 0) {
+        element.warnings.push({
+          message: `Unable to resolve superclass ${scannedElement.superClass}`,
+          severity: Severity.ERROR,
+          code: 'unknown-superclass',
+          // TODO(justinfagnani): we need to know where the superclass came
+          // (annotation vs extends clause) from and track its range.
+          sourceRange: null as any as SourceRange,
+        });
+      }
+    }
+  }
 
-  const flatteningResult = getFlattenedAndResolvedBehaviors(
-      scannedElement.behaviorAssignments, document);
+  // TODO(justinfagnani): Copy over all properties better, or have
+  // PolymerElement wrap ScannedPolymerElement.
+  element.abstract = scannedElement.abstract;
+  element.astNode = scannedElement.astNode;
+  scannedElement.attributes.forEach((o) => element.attributes.push(o));
+  scannedElement.behaviorAssignments.forEach(
+      (o) => element.behaviorAssignments.push(o));
+  element.className = scannedElement.className;
+  scannedElement.demos.forEach((o) => element.demos.push(o));
+  element.description = scannedElement.description;
+  element.domModule = scannedElement.domModule;
+  scannedElement.events.forEach((o) => element.events.push(o));
+  element.extends = scannedElement.extends;
+  element.jsdoc = scannedElement.jsdoc;
+  scannedElement.listeners.forEach((o) => element.listeners.push(o));
+  scannedElement.observers.forEach((o) => element.observers.push(o));
+  scannedElement.properties.forEach((o) => element.properties.push(o));
+  element.scriptElement = scannedElement.scriptElement;
+  scannedElement.slots.forEach((o) => element.slots.push(o));
+  element.sourceRange = scannedElement.sourceRange!;
+  element.superClass = scannedElement.superClass;
+  element.tagName = scannedElement.tagName;
+  scannedElement.warnings.forEach((o) => element.warnings.push(o));
+
+  //
+  // Behaviors
+  //
+  const behaviorsAndWarnings =
+      getBehaviors(scannedElement.behaviorAssignments, document);
 
   // This has the combined effects of copying the array of warnings from the
   // ScannedElement, and adding in any new ones found when resolving behaviors.
-  clone.warnings = clone.warnings.concat(flatteningResult.warnings);
-
-  const behaviors = Array.from(flatteningResult.resolvedBehaviors);
-  clone.properties = mergeByName(
-      scannedElement.properties,
-      behaviors.map((b) => ({name: b.className, vals: b.properties})));
-  clone.attributes = mergeByName(
-      scannedElement.attributes,
-      behaviors.map((b) => ({name: b.className, vals: b.attributes})));
-  clone.events = mergeByName(
-      scannedElement.events,
-      behaviors.map((b) => ({name: b.className, vals: b.events})));
+  element.warnings = element.warnings.concat(behaviorsAndWarnings.warnings);
+
+  const behaviors = Array.from(behaviorsAndWarnings.behaviors);
+
+  // for (const behavior of behaviors) {
+  //   inheritFrom(element, behavior);
+  // }
```

Comment:
Added a TODO - I want to handle superclasses, behaviors and soon mixins more similarly.